sdk/ai/azure-ai-inference/README.md

```diff
@@ -103,6 +103,25 @@ client = ChatCompletionsClient(
 
 During application development, you would typically set up the environment for authentication using Entra ID by first [Installing the Azure CLI](https://learn.microsoft.com/cli/azure/install-azure-cli), running `az login` in your console window, then entering your credentials in the browser window that was opened. The call to `DefaultAzureCredential()` will then succeed. Setting `exclude_interactive_browser_credential=False` in that call will enable launching a browser window if the user isn't already logged in.
 
+### Defining default settings while creating the clients
+
+The method `with_defaults` can be called on the clients, in order to define default input settings that will apply to all future service calls.
+
+For example, here we create a `ChatCompletionsClient` using API key authentication, and apply two settings, `temperature` and `max_tokens`:
+
+```python
+from azure.ai.inference import ChatCompletionsClient
+from azure.core.credentials import AzureKeyCredential
+
+client = ChatCompletionsClient(
+    endpoint=endpoint,
+    credential=AzureKeyCredential(key)
+).with_defaults(
+    temperature=0.5,
+    max_tokens=1000
+)
```

Comment:
Wondering if these defaults should just be keyword-only arguments on the client instead of being set through a `with_defaults` helper. I haven't seen this pattern in any other Python libraries. @johanste for his thoughts.

Reply:
The parameters should be available in the constructor as well. We don't have this pattern in our other azure libraries today, mainly because we have a wider set of operations with different parameters.

Reply:
Furthermore, we should consider the lifetime of the client(s) involved when used in a context manager. This example is problematic as is because it doesn't show which client(s) need to be closed. In CPython/with a sync client, that is less problematic. But for async, you need to be careful.

Reply:
With latest commits, the constructor exposes the chat completions settings as hidden keyword arguments on the constructor. I do not plan to publicly document them. They are there only to support implementation of `.with_defaults()`. Similarly for the two Embeddings clients.
With regards to the other issue -- async clients and context -- I'm waiting for you @johanste to clarify what package README and/or samples changes are required, as after additional review I do not see a need for changes.